version: '3.9'

services:
  db:
    image: postgres
    volumes:
      - ./data/db:/var/lib/postgresql/data/
    environment:
      - POSTGRES_DB=postgres
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
    ports:
      - "5432:5432"
    restart: on-failure
  redis:
    image: redis:alpine
  celery:
    build: .
    command: celery -A project worker -l info
    volumes:
      - .:/app/
    environment:
      - POSTGRES_DB=postgres
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
      - DEBUG=1
      - SECRET_KEY=adsds1456%*3r9-=z-+_mz4r-!qeed@(-a_r(g@kjo8y3r27%m
      - DJANGO_ALLOWED_HOSTS=localhost 127.0.0.1 [::1]
    depends_on:
      #- db
      - redis
  celery-beat:
    build: .
    command: celery -A project beat -l info
    volumes:
      - .:/app/
    environment:
      - POSTGRES_DB=postgres
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
      - DEBUG=1
      - SECRET_KEY=adsds1456%*3r9-=z-+_mz4r-!qeed@(-a_r(g@kjo8y3r27%m
      - DJANGO_ALLOWED_HOSTS=localhost 127.0.0.1 [::1]
    depends_on:
      #- db
      - redis
  web:
    build: .
    ports:
      - 8000:8000
    environment:
       - POSTGRES_DB=postgres
       - POSTGRES_USER=postgres
       - POSTGRES_PASSWORD=postgres
       - SECRET_KEY=adsds1456%*3r9-=z-+_mz4r-!qeed@(-a_r(g@kjo8y3r27%m
    entrypoint: /app/docker-entrypoint.sh
    volumes:
      - .:/app/
    depends_on:
       - db
       - redis
    restart: on-failure